server/controllers/users.js

```diff
@@ -0,0 +1,67 @@
+'use strict';
+
+const atob = require('atob');
+
+
+const userModel = require('../models/users.js');
+
+exports.login = function* (next) {
+  try {
+    // sample: cnViZW46MTIzNA==
+    let auth = this.request.header.authorization.split(' ')[1];
+    auth = atob(auth).split(':');
+
+    let userInfo = yield new Promise(function(resolve,reject) {
```

Comment:
Actually, from this line to 23 can be done all together with a `user.getWithCredentials(auth[0], auth[1])`. If you the function returns anything, then the user has authenticated himself.